### ISSUE TITLE
Use ST_AsMVT instead of Mapnik for tile generation

### ISSUE THREAD

**Issue by nyurik** [openmaptiles/openmaptiles#875]
There are now more and more issues showing up with the outdated and unmaintained Mapnik library that impacts OpenMapTiles project:

* We do not have any reasonable tests for Mapnik tile generation, nor can we measure the tile generation performance (per tile speed and size). We can only test it with `ST_AsMVT`.
* Mapnik hides some geometries (#832 - breaks production https://github.com/IsraelHikingMap/openmaptiles/issues/21)

| Tileserver (Mapnik) | Postserve (ST_AsMVT) |
|---|---|
| <img width="157" alt="2020-04-26 12_09_02-2020-04-26 11_26_21-Ubunto - 31 154 215 180_3389 - Remote Desktop Connection" src="https://user-images.githubusercontent.com/1304610/80303199-6155cf80-87b7-11ea-9342-84a3d8b25fdf.png"> | <img width="157" alt="2020-04-26 12_09_06-2020-04-26 11_25_40-Ubunto - 31 154 215 180_3389 - Remote Desktop Connection" src="https://user-images.githubusercontent.com/1304610/80303208-72064580-87b7-11ea-86e3-0950b10927d2.png"> |

* Mapnik produces a lot of errors and warnings that require complex workaround just to keep the output sane (https://github.com/openmaptiles/openmaptiles-tools/pull/256)
* Not being maintained means there are security bugs that are not being addressed.
* Mapnik does not scale well, requiring tile generation to have two server farms -- one for PostgreSQL, and one for Mapnik. The `ST_AsMVT` approach uses just the PostgreSQLs - the more servers it has, the faster the generation.
* Mapnik requires us to maintain a lot of extra code to generate Mapnik configuration files

This is **not** an urgent matter, but we should establish a migration timeline to remove this obsolete unmaintained component from the pipeline.

cc: @klokan @zstadler @daliborjanak @MartinMikita @TomPohys

**Comment by zstadler** [comment]
For me, it **is** becoming an urgent matter, as I'm unable to find a solution or workaround the bug in #832

**Comment by daliborjanak** [comment]
We have a same problems so scheduled the work on this in August. We will inform and discuss the steps when we start on it

**Comment by zstadler** [comment]
@nyurik  @klokan @daliborjanak @MartinMikita @TomPohys
I'd like to start a draft PR for this issue.
What would be the preferred migration path: a PR that keeps using the existing `make generate-tiles` and `docker-compose generate-tiles` and changes the underlying implementation, or a PR that adds a new make target and a supporting docker-compose service?

**Comment by klokan** [comment]
Great @zstadler!

I suggest to make it independent - to be able to choose the approach (mapnik vs st_asmvt)
Perhaps `make generate-tiles-pg` and new `docker-compose generate-vectortiles-pg`?

To be true - I think we have a chance to get rid of tilelive dependecy here as well - and make a new tool directly in openmaptiles-tools itterating on the tilegrid and zooms - to be able to support also non-Mercator coordinate systems.

The postgres database itself could define the tilegrid in a table (zoomlevels, tile ranges and resolutions for each zoom levels) - with default / if undefined - hardcoded to Mercator.

This way the generator / real time server could directly respond with a tile.

Comments welcome.

**Comment by nyurik** [comment]
I don't think we need a separate docker compose service for this. Unlike the current approach, we can easily add the needed components to the main tools docker image, and so generation will be no different than calling any other tools command.  The only possible DC service we may want is for the updater (using the same tools image)

**Comment by TomPohys** [comment]
The new makefile target `generate-tiles-pg` was added by https://github.com/openmaptiles/openmaptiles/pull/1048

**Comment by nyurik** [comment]
I just had a chance to revisit this issue. It seems we are still using Mapnik as the **default** quickstart target, which I believe we should not do. It is great to support multiple ways to render, but quickstart should point towards the most robust and maintained one.  Also, it seems our tools repo no longer builds due to mapnik failing to compile. This adds significant burden to maintain the tooling -- the last tools release is over 9 months old.  @klokan are you sure you want to continue maintaining Mapnik support despite that project failing further and further behind, as well as it generating bugs that noone seems to be fixing like #832 ?

**Comment by mohammad-babaei** [comment]
When i use the legacy Mapnik library it takes seconds to generate the mbtiles but when i use the new method using ST_AsMVT it says it takes about 65 days!!!
How can i improve that?
